lib/rucio/core/did.py

```diff
@@ -692,7 +694,19 @@ def delete_dids(dids, account, expire_rules=False, session=None, logger=logging.
         with record_timer_block('undertaker.parent_content'):
             for parent_did in session.query(models.DataIdentifierAssociation).filter(or_(*parent_content_clause)):
                 existing_parent_dids = True
-                detach_dids(scope=parent_did.scope, name=parent_did.name, dids=[{'scope': parent_did.child_scope, 'name': parent_did.child_name}], session=session)
+                detach_dids(scope=parent_did.scope, name=parent_did.name, dids=[{'scope': parent_did.child_scope, 'name': parent_did.child_type}], session=session)
+
+    # Set Epoch tombstone for the files replicas inside the did
+    if config_core.get('undertaker', 'remove_all_files', default=False, session=session) and file_content_clause:
```

Comment:
remove_all_files I find not very clear as an option, I think we should find another name.

Reply:
Changed to `purge_all_replicas`